x-pack/plugins/lens/public/state_management/selectors.ts

```diff
@@ -129,6 +132,14 @@ export const selectSavedObjectFormat = createSelector(
       });
     });
 
+    let persistibleVisualizationState = visualization.state;
+    if (activeVisualization.getPersistableState) {
+      const { state: persistableState, savedObjectReferences } =
+        activeVisualization.getPersistableState(visualization.state);
+      persistibleVisualizationState = persistableState;
+      references.push(...savedObjectReferences);
```

Comment:
Doesn't work for ad-hoc data views, this needs the same logic as the datasource state: https://github.com/elastic/kibana/blob/01e8881decc9fd785e2589d938c8a5f03c36f4e5/x-pack/plugins/lens/public/state_management/selectors.ts#L127